### PR TITLE
refactor gridlinesTests

### PR DESCRIPTION
**Pull request by ztsai**
screen shot 2015-11-12 at 1 24 15 pm

test clean up
part of #2625

### CODE REVIEW

File: test/components/gridlinesTests.ts [hunk 1]
```diff
@@ -1,62 +1,62 @@
...
-    let xScale = new Plottable.Scales.Linear();
+  const SVG_WIDTH = 640;
+  const SVG_HEIGHT = 480;
+  let svg: d3.Selection<void>;
+  let xScale: Plottable.Scales.Linear;
+  let yScale: Plottable.Scales.Linear;
+  let gridlines: Plottable.Components.Gridlines;
+
+  beforeEach(() => {
+    svg = TestMethods.generateSVG(640, 480);
```
Comment (jtlan) [inline test/components/gridlinesTests.ts]: Should probably use the `const`s here, but I don't think this test is even dependent on the dimensions, so we can probably go with the defaults.

File: test/components/gridlinesTests.ts [hunk 2]
```diff
@@ -1,62 +1,62 @@
...
+  });
 
-    let gridlines = new Plottable.Components.Gridlines(xScale, yScale);
+  it("sets ranges of scales to the Gridlines dimensions when layout is computed", () => {
     gridlines.renderTo(svg);
 
-    assert.deepEqual(xScale.range(), [0, 640], "x scale range extends to the width of the svg");
-    assert.deepEqual(yScale.range(), [480, 0], "y scale range extends to the height of the svg");
+    assert.deepEqual(xScale.range(), [0, SVG_WIDTH], "x scale range extends to the width of the svg");
+    assert.deepEqual(yScale.range(), [SVG_HEIGHT, 0], "y scale range extends to the height of the svg");
```
Comment (jtlan) [inline test/components/gridlinesTests.ts]: If we get rid of `SVG_WIDTH` and `SVG_HEIGHT`, probably use `numAttr(svg, ...)` here.

File: test/components/gridlinesTests.ts [hunk 3]
```diff
@@ -1,62 +1,62 @@
...
 
     svg.remove();
   });
 
-  it("Gridlines and axis tick marks align", () => {
-    let svg = TestMethods.generateSVG(640, 480);
-    let xScale = new Plottable.Scales.Linear();
-    xScale.domain([0, 10]); // manually set domain since we won't have a renderer
+  it("aligns Gridlines and axis tick marks", () => {
```
Comment (jtlan) [inline test/components/gridlinesTests.ts]: This test is probably more accurately "uses the TickGenerators of its Scales" -- that's what's actually going on, and that way we don't need to make Axes.

**Comment by jtlan** [comment]
Can we add some tests that the `Gridlines` update when its `Scale`s update?

File: test/components/gridlinesTests.ts [hunk 4]
```diff
@@ -1,63 +1,116 @@
...
+    assert.deepEqual(yScale.range(), [TestMethods.numAttr(svg, "height"), 0], "y scale range extends to the height of the svg");
 
     svg.remove();
   });
 
-  it("Gridlines and axis tick marks align", () => {
-    let svg = TestMethods.generateSVG(640, 480);
-    let xScale = new Plottable.Scales.Linear();
-    xScale.domain([0, 10]); // manually set domain since we won't have a renderer
+  it("uses the TickGenerators of its Scales", () => {
```
Comment (jtlan) [inline test/components/gridlinesTests.ts]: Can you change this test to not use `Axes`? Basically, we should be able to render the `Gridlines`, then check that the lines are in the places specified by `xScale.ticks()` and `yScale.ticks()`.

File: test/components/gridlinesTests.ts [hunk 5]
```diff
@@ -1,63 +1,116 @@
...
+      assert.closeTo(y, yScale.scale(yTicks[i]), 1, "y gridline is updated");
+    });
+    svg.remove();
+  });
+
+  it("throws error on non-Quantitative Scales", () => {
+    const categoryScale = new Plottable.Scales.Category();
+    // HACKHACK #2661: Cannot assert errors being thrown with description
+    (<any> assert).throw(() => new Plottable.Components.Gridlines(<any> categoryScale, null), Error,
+      "xScale needs to inherit from Scale.QuantitativeScale", "can't not set xScale to category scale");
```
Comment (jtlan) [inline test/components/gridlinesTests.ts]: Probably "can't" instead of "can't not"?

Comment (ztsai) [inline test/components/gridlinesTests.ts]: LOL

File: test/components/gridlinesTests.ts [hunk 6]
```diff
@@ -1,63 +1,116 @@
...
+
+  it("updates gridlines when scales update", () => {
+    gridlines.renderTo(svg);
+
+    let xGridlines = gridlines.content().select(".x-gridlines").selectAll("line");
+    let xTicks = xScale.ticks();
+    assert.strictEqual(xGridlines.size(), xTicks.length, "There is an x gridline for each x tick");
+    xGridlines.each(function(gridline, i) {
+      const x = TestMethods.numAttr(d3.select(this), "x1");
+      assert.closeTo(x, xScale.scale(xTicks[i]), 1, "x gridline drawn on ticks");
```
Comment (jtlan) [inline test/components/gridlinesTests.ts]: This is the logic that should be in the "uses tick generators" test.

Also, we should use either `strictEqual`, or `closeTo` with `window.Pixel_CloseTo_Requirement`.

**Pull request merged**